src/hooks/useSelectImageMenu.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useCallback } from 'react';
+import ImagePicker, { Image, Options } from 'react-native-image-crop-picker';
+import { ContextMenuButton } from 'react-native-ios-context-menu';
+import { useMutation } from 'react-query';
+import {
+  uploadImage,
+  UploadImageReturnData,
+} from '@rainbow-me/handlers/pinata';
+import { showActionSheetWithOptions } from '@rainbow-me/utils';
+
+type Action = 'library' /* | 'nft'*/;
+
+const items = {
+  library: {
+    actionKey: 'library',
+    actionTitle: 'Upload photo',
+    icon: {
+      imageValue: {
+        systemName: 'photo',
+      },
+      type: 'IMAGE_SYSTEM',
+    },
+  },
+  // nft: {
+  //   actionKey: 'nft',
+  //   actionTitle: 'Choose NFT',
+  //   icon: {
+  //     imageValue: {
+  //       systemName: 'cube',
+  //     },
+  //     type: 'IMAGE_SYSTEM',
+  //   },
+  // },
+};
+
+export default function useSelectImageMenu({
+  imagePickerOptions,
+  menuItems = ['library'],
+  onChangeImage,
+  onUploading,
+  onUploadSuccess,
+  onUploadError,
+  uploadToIPFS = false,
+}: {
+  imagePickerOptions?: Options;
+  menuItems?: Action[];
+  onChangeImage?: ({ image }: { image: Image }) => void;
+  onUploading?: ({ image }: { image: Image }) => void;
+  onUploadSuccess?: ({
+    data,
+    image,
+  }: {
+    data: UploadImageReturnData;
+    image: Image;
+  }) => void;
+  onUploadError?: ({ error, image }: { error: unknown; image: Image }) => void;
+  uploadToIPFS?: boolean;
+} = {}) {
+  const { isLoading: isUploading, mutateAsync: upload } = useMutation(
+    'ensImageUpload',
+    uploadImage
+  );
+
+  const handleSelectImage = useCallback(async () => {
+    const image = await ImagePicker.openPicker({
+      ...imagePickerOptions,
+      includeBase64: true,
+      mediaType: 'photo',
+    });
+    onChangeImage?.({ image });
+
+    if (uploadToIPFS) {
```

Comment:
i was trying this and I think we don't want to upload the images every time the user selects one, if the image is changed many times or if the user sets an image but doesn't register the ENS we'll upload bunch of images that won't be used

we probably want to upload just before sending the multicall tx to send the tx with the image link, WDYT?

Reply:
yeah thats true. maybe we can add this as a note for now, and then implement once we have the designs for the multicall tx finalised?